pkg/backend/kube/tootles.go

```diff
@@ -110,6 +111,294 @@ func toEC2Instance(hw v1alpha1.Hardware) data.Ec2Instance {
+		i.Userdata = *hw.Spec.UserData
+	}
+
+	// Generate network configuration from Hardware resource
+	i.NetworkConfig = generateNetworkConfigV2(hw)
+
+	return i
+}
+
+// getNameServers extracts nameservers from Hardware interfaces.
+// Returns IPv4 and IPv6 nameservers separately.
+func getNameServers(hw v1alpha1.Hardware) (ipv4DNS []string, ipv6DNS []string) {
+	// Try to get nameservers from the first interface with DHCP config
+	for _, iface := range hw.Spec.Interfaces {
+		if iface.DHCP != nil && len(iface.DHCP.NameServers) > 0 {
+			// Separate IPv4 and IPv6 nameservers
+			for _, ns := range iface.DHCP.NameServers {
+				if strings.Contains(ns, ":") {
+					ipv6DNS = append(ipv6DNS, ns)
+				} else {
+					ipv4DNS = append(ipv4DNS, ns)
+				}
+			}
+			break
+		}
+	}
+
+	return ipv4DNS, ipv6DNS
+}
+
+func cidrFromNetmask(netmask string) string {
+	if netmask == "" {
+		return ""
+	}
+
+	parts := strings.Split(netmask, ".")
+	if len(parts) != 4 {
+		return ""
+	}
+
+	setBits := 0
+	for _, part := range parts {
+		octet := 0
+		for _, ch := range part {
+			if ch < '0' || ch > '9' {
+				return ""
+			}
+			octet = octet*10 + int(ch-'0')
+		}
+
+		if octet < 0 || octet > 255 {
+			return ""
+		}
+
+		for octet > 0 {
+			setBits++
+			octet &= (octet - 1)
+		}
+	}
+
+	if setBits < 0 || setBits > 32 {
+		return ""
+	}
+
+	return fmt.Sprintf("%d", setBits)
+}
+
+// generateNetworkConfigV2 creates a NoCloud-compatible network configuration (version 2) from Hardware resource.
+// Version 2 is the modern Netplan-compatible format.
+// Only generates configuration for network bonding. For non-bonded interfaces, cloud-init handles default DHCP.
+// Returns nil if no network configuration is needed (non-bonded interfaces use default DHCP).
+func generateNetworkConfigV2(hw v1alpha1.Hardware) *data.NetworkConfig {
+	networkSpec := data.NetworkSpecV2{
+		Version: 2,
+	}
+
+	// Check if bonding is enabled
+	bondingEnabled := hw.Spec.Metadata != nil && hw.Spec.Metadata.BondingMode > 0
+
+	if bondingEnabled && len(hw.Spec.Interfaces) >= 2 {
+		// Generate bonding configuration
+		ethernets, bonds := generateBondingConfigurationV2(hw)
+		networkSpec.Ethernets = ethernets
+		networkSpec.Bonds = bonds
+	} else {
+		// No bonding configuration needed, return nil to use default DHCP
+		return nil
+	}
+
+	return &data.NetworkConfig{
+		Network: networkSpec,
+	}
+}
+
+// generateBondingConfigurationV2 creates bonding configuration (v2 format) from Hardware resource.
+// Returns separate ethernets and bonds maps.
+// Requires MAC addresses for all interfaces to enable proper matching.
+func generateBondingConfigurationV2(hw v1alpha1.Hardware) (map[string]data.EthernetConfig, map[string]data.BondConfig) {
+	ethernets := map[string]data.EthernetConfig{}
+	bonds := map[string]data.BondConfig{}
+	bondInterfaces := []string{}
+
+	// Create physical interfaces for bonding (without IP addresses)
+	phyIndex := 0
+	for _, iface := range hw.Spec.Interfaces {
```

Comment:
If i understand this correctly, then all interfaces defined in the Hardware object will be included in the bond configuration. Is that correct? Is that the expected behavior? If so, I think there needs to be some consideration to interfaces that are defined but should not be included in the bond.